### PR TITLE
[FEAT] [no-this-alias] Add rule

### PR DESCRIPTION
**Pull request by j-f1**

### CODE REVIEW

File: lib/rules/no-this-alias.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            },
+        ],
+    },
+
+    create(context) {
+        const { allowDestructuring = false, allowedNames = [] } =
+            context.options[0] || {};
+
+        return {
+            VariableDeclarator(node) {
```
Comment (armano2) [inline lib/rules/no-this-alias.js]:
```suggestion
            "VariableDeclarator[kind!='type']"(node) {
```

do you mind adding check if it's not type alias?

**Comment by armano2** [comment]
@j-f1 you have to update `yarn.lock` you are using `eslint-docs@0.2.6` instead of `eslint-docs@0.2.3`

**Comment by j-f1** [comment]
Done @armano2.

File: lib/rules/no-this-alias.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                if (init.type !== "ThisExpression") return;
+                if (allowDestructuring && node.id.type !== "Identifier") return;
+
+                if (!allowedNames.includes(id.name)) {
+                    context.report({
+                        node: id,
+                        message:
+                            id.type === "Identifier"
+                                ? "Unexpected aliasing of 'this' to local variable"
+                                : "Unexpected aliasing of members of 'this' to local variables",
```
Comment (bradzacher) [inline lib/rules/no-this-alias.js]: could you please make use of `messageId` instead?

**Comment by j-f1** [comment]
Done @bradzacher

**Pull request merged**